src/twisted/web/server.py

```diff
@@ -525,7 +525,8 @@ def encoderForRequest(self, request):
         """
         acceptHeaders = request.requestHeaders.getRawHeaders(
             'accept-encoding', [])
-        supported = ','.join(acceptHeaders).split(',')
+        supported = [encoding.strip()
+                     for encoding in ','.join(acceptHeaders).split(',')]
```

Comment:
This join/split dance has to allocate roughly `sum((len(h) for h in acceptHeaders)) + len(acceptHeaders) - 1` and then just throws it away.

In practice this is limited to something like 8kb because the total twisted request size is 16k with a maximum of 500 headers, so it's not like a single request is likely to exhaust all your memory by sending the maximum number of `Accept-Encoding` headers. But it's still a lot of malloc and memcpy.

Some simple loops are probably better?

```python
for header in acceptHeaders:
    for encoding in header.split(','):
        if 'gzip' in encoding.strip():
           encoding = ...
           break
```

Of course, split and strip are also going to do a lot of copying, so maybe a regexp instead?

Something along the lines of 
```python
gzipRe = re.compile(r'[\s,]?gzip[\s,]?')
```

and 
```python
for header in acceptHeaders:
    if gzipRe.match('header'):
        encoding = ...
        break
```